javarosa/src/main/java/org/javarosa/xpath/expr/XPathFuncExpr.java

```diff
@@ -1449,7 +1449,7 @@ public static Object InferType(String attrValue) {
         }
 
         try {
-            Double ret = new Double(Double.parseDouble(attrValue));
+            Double ret = MathUtils.parseDoubleSafe(attrValue);

```

Comment:
The results of this evaluation are cached, so hopefully that would help soften any adverse performance effects. Would be cool to test this vs. the infinity check on a few thousand unique ID strings though.